### PR TITLE
Database

### PR DESCRIPTION
**Pull request by 4rt3ry**
## Database

Convert all data storage from global variables and local files to PostgreSQL. 

## Database Setup
The database is automatically set up the first time the application is started using `sequelize.sync()`. 

## Commands that use the database

`/remember` and all of its sub commands
`/check-in` and  all of its sub commands
`/availability` and all of its sub commands
`/set-availability-channel`
`/summarize`

## Additional files that use the database
`events/interactionCreate/checkInSubmit.js`
`events/interactionCreate/validateAval.js`
`events/interactionCreate/validateUnAval.js`
`events/interactionCreate/removeSchedule.js`
`events/messageCreate/*.js`
`events/ready/checkInSchedule.js`
`events/ready/unavailableAssign.js`

## Known Issues

I have done basic tests for every command and have not found any issues. Note that there might still be unaddressed problems  caused by edge cases. A good number of errors were caused by missing `await` statements, meaning any future bugs are likely to be the same.

### CODE REVIEW

**Review by tstodter — COMMENTED** [review]
I think we're missing [`model.sync`](https://sequelize.org/docs/v6/core-concepts/model-basics/#model-synchronization), or migrations to get the tables actually made. On a fresh local database, just running this branch now, I immediately get `relation "checkin_reminders" does not exist` and `relation "unavailable_schedules" does not exist`, and checking the database itself there's no tables made. I'm guessing you've already got them in your local db from the dev process.

Suggest we start with .sync; I can follow up with proper migrations.

File: example.env [hunk 1]
```diff
@@ -1,4 +1,11 @@
 DISCORD_TOKEN='example bot token here'
 TESTSERVER_ID= 'example server id here'
 DEV_IDS= ['dev id here', 'dev id here']
+DATABASE_URL=postgres://user:password@host:port/database
+PGUSER=username
+PGHOST=host
+PGPASSWORD=password
+PGDATABASE=database_name
+PGPORT=5432
```
[the remark this answers is not in the document]
Comment (4rt3ry) [reply example.env:9]: I removed the individual entries in favor of DATABASE_URL. They can't be used with Heroku Postgres anyway.

File: src/database/setup.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+const models = require('./models');
+
+/**
+ * Call this function once in src/index.js to set up a new database
+ */
+const setup = async (force = false) => {
+    models.User.sync({ force });
```
Comment (tstodter) [inline src/database/setup.js:8]: Why not just run this every time when the server starts, with force = false?

File: src/database/models.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        ...sslOptions,
+        pool:     {
+            max:     5,
+            min:     0,
+            acquire: 30000,
+            idle:    10000
+        }
+    }
+);
+
```
Comment (tstodter) [inline src/database/models.js:25]: Would suggest injecting the sequelize object into this file; basically, turn the whole file into one big function that accepts the `sequelize` instance object as a parameter, and internally calls `sequelize.define` for all the models.

This is kind of what I mean, generally: https://github.com/sequelize/express-example/blob/7304dc6ff4b842b06301d962bd0bcb07454042b8/express-main-example/sequelize/index.js

So then we'd have a separate file that did the work above and exported the sequelize instance. Somewhere near the start of the server, we import that file to get the sequelize instance, and pass it into this new `apply and make the models` function. Probably call `sequelize.sync()` then. At that point, we have a handle on the sequelize instance near the top of the program (and can pass it down to where we need it), and can access the models on it with `sequelize.models.Whatever`.

File: src/database/queries.js [hunk 4]
```diff
@@ -0,0 +1,912 @@
+const path = require('path');
+require('dotenv').config({ path: path.resolve(__dirname, '../.env') });
+
+const { Pool } = require('pg');
+
```
Comment (tstodter) [inline src/database/queries.js:5]: This pool stuff in this file looks like it might be leftover junk?

File: src/database/queries.js [hunk 5]
```diff
@@ -0,0 +1,912 @@
...
+    // TODO: figure out a way to filter out past dates
+    // TODO: figure out a way to delete past schedules
+
+    return getModelEntryByUser(discord_user_id, AvailableSchedule);
+
+    // return getUser(discord_user_id).then(user => {
+    //     const filter = { where: { user_id: user.id } };
+    //     return AvailableSchedule.findOne(filter);
+    // })
```
Comment (tstodter) [inline src/database/queries.js:827]: Some junk like these comments in this file. If it's a time sink then skip cleaning it up, but I'll want to follow up with a cleanup PR.

File: src/database/setup.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    models.AvailableSchedule.sync({ force });
+    models.CheckInReminder.sync({ force });
+    models.CheckInResponse.sync({ force });
+    models.CheckInSchedule.sync({ force });
+    models.Config.sync({ force });
+    models.Message.sync({ force });
+    models.UnavailableSchedule.sync({ force });
+    models.UnavailableStart.sync({ force });
+    models.UnavailableStop.sync({ force });
```
Comment (tstodter) [inline src/database/setup.js:17]: Can probably just say `sequelize.sync()`.
